system/lib/wasmfs/wasmfs.cpp

```diff
@@ -76,19 +77,80 @@ __wasi_errno_t __wasi_fd_write(__wasi_fd_t fd,
 
   auto lockedFile = file->locked();
 
-  __wasi_size_t num = 0;
+  off_t offset = lockedOpenFile.position();
```

Comment:
I think I called it position elsewhere because one of the input parameters was also `offset`. Maybe `newOffset`, even though we would have to use the `new` here.